File: Expenses/Expenses.ApiRepository/Repositories/Categories.cs
using Expenses.ApiRepository.Interfaces;
using Expenses.Common;
using Expenses.Model.Dto;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Http;
using System.Text;
using System.Threading.Tasks;

namespace Expenses.ApiRepository.Repositories
{
    class Categories : RepositoryBase, ICategories
    {
        public Categories(string host, string path, string apiRepositoryName = "categories") 
            : base(host, path, apiRepositoryName)
        {
        }

        public async Task<HttpResponseMessage> GetAll(string login, string key)
        {
            var uri = BaseUri.Append("getall", login);
            using (var httpClient = new HttpClient())
            {
                httpClient.DefaultRequestHeaders.Add("x-functions-key", key);
                return await httpClient.GetAsync(uri);
            }
        }

        public async Task<HttpResponseMessage> Add(string login, string key, string categoryName, Dictionary<string,double> factor = null)
        {
            var uri = BaseUri.Append("getall", login);
            var dto = new AddCategoryDto()
            {
                Name = categoryName,
                Factor = factor
            };
            var content = new StringContent(JsonConvert.SerializeObject(dto));
            using (var httpClient = new HttpClient())
            {
                httpClient.DefaultRequestHeaders.Add("x-functions-key", key);
                return await httpClient.PostAsync(uri, content);
            }
        }
    }
}

File: Expenses.AndroidClient/Expenses.Repository/src/main/java/net/azurewebsites/expenses_by_klaudia/repository/Categories.java
package net.azurewebsites.expenses_by_klaudia.repository;

import net.azurewebsites.expenses_by_klaudia.model.AddCategoryDto;

import java.net.HttpURLConnection;
import java.util.Dictionary;
import java.util.HashMap;

public class Categories extends RepositoryBase {
    public Categories(String host, String path) {
        super(host, path, "categories");
    }

    public HttpURLConnection GetAll(String login, String key) throws RepositoryException {
        String uri = GetBaseUriBuilder()
                .appendPath("getall")
                .appendPath(login)
                .appendQueryParameter("code", key)
                .build()
                .toString();
        return SendGet(uri);
    }

    public HttpURLConnection Add(String login, String key, String categoryName, HashMap<String, Double> factors) throws RepositoryException {
        String uri = GetBaseUriBuilder()
                .appendPath("add")
                .appendPath(login)
                .appendQueryParameter("code", key)
                .build()
                .toString();
        AddCategoryDto dto = new AddCategoryDto();
        dto.Name = categoryName;
        dto.Factor = factors;
        return SendJsonPost(uri, dto);
    }
}
